src/components/table/index.tsx

```diff
@@ -1,46 +1,189 @@
 import {
   Box,
+  IconButton,
+  Stack,
   Table,
   TableBody,
   TableCell,
   TableContainer,
   TableRow
 } from '@mui/material';
+import DeleteIcon from '@mui/icons-material/Delete';
+import EditIcon from '@mui/icons-material/Edit';
 import CustomTableHead from './table-head';
-import { Column, columns } from '../../models/user';
+import { Column, UpdateUserPayload } from '../../models/user';
 import { HEADER_HEIGHT } from '../../constants/common';
-import { useQuery } from '@tanstack/react-query';
-import { admin } from '../../api';
-import React from 'react';
-
-const CustomTable = () => {
-  const { data, isLoading } = useQuery({
-    queryKey: ['getAllUsers'],
-    queryFn: async () => await admin.getAllUsers()
+import {
+  QueryClient,
+  useMutation,
+  useQuery,
+  useQueryClient
+} from '@tanstack/react-query';
+import { deleteUser, getAllUsers } from '../../api/admin';
+import React, { useState } from 'react';
+import AddUserDialog from '../dialogs/add-user-dialog';
+import ConfirmDialog from '../dialogs/confirm-dialog';
+import { ToastMessage } from '../toast';
+import { toast } from 'react-toastify';
+import TablePagination from './table-pagination';
+import { TableState } from '../../models/table';
+import { debounce } from 'lodash';
+interface CustomTableProps {
```

Comment:
Please change all `interface` to `type`

Case to use `interface` should be inheritance and abstract stuff, if don't use type instead